kart/structure.py

```diff
@@ -235,6 +236,14 @@ def create_tree_from_diff(
                 object_builder.insert(path, blob)
 
         for ds_path, ds_diff in repo_diff.items():
+            if ds_path == FILES_KEY:
+                self.apply_file_diff(
```

Comment:
this method should maybe be `apply_files_diff` since it's not just _one_ file, it's all the files